oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoTestClient.java

```diff
@@ -16,31 +16,136 @@
  */
 package org.apache.jackrabbit.oak.plugins.document.mongo;
 
+import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 
-import com.mongodb.MongoClient;
-import com.mongodb.MongoClientURI;
+import com.mongodb.ClientSessionOptions;
+import com.mongodb.ConnectionString;
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.ListDatabasesIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
 import com.mongodb.client.MongoDatabase;
-
+import com.mongodb.client.MongoIterable;
+import com.mongodb.connection.ClusterDescription;
+import org.apache.jackrabbit.oak.plugins.document.util.MongoConnection;
+import org.bson.Document;
+import org.bson.conversions.Bson;
 import org.jetbrains.annotations.NotNull;
 
-class MongoTestClient extends MongoClient {
+class MongoTestClient implements MongoClient {
```

Comment:
This is now an interface and I have found no better way to achieve the same result.

Reply:
The only alternative that comes to my mind is using `java.lang.reflect.Proxy`, but implementing the methods is also fine with me.